datafusion/core/src/execution/session_state.rs

```diff
@@ -1234,3 +1236,34 @@ impl<'a> SimplifyInfo for SessionSimplifyProvider<'a> {
         expr.get_type(self.df_schema)
     }
 }
+
+/// The state store that stores the reference of the runtime session state.
+pub struct StateStore {
```

Comment:
I moved `StateStore` to `session_state.rs` because I think it could be used by others that has similar scenario with `DynamicFileSchemaProvider`